### PR TITLE
Refactor/use cohort to determine what is new lesson pages - LESQ-524

### PR DESCRIPTION
**Pull request by JBR90**
## Description

Music year: 1978

- Adds cohort to lesson overview
- Filters out legacy data a query level
- Changes 'new' tag and if we show share button from cohort.

## How to test

1. Go to https://deploy-preview-2214--oak-web-application.netlify.thenational.academy
Lesson overview page - everything should work as before.
## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOveview.gql [hunk 1]
```diff
@@ -2,12 +2,14 @@ query lessonOverview(
   $programmeSlug: String!
   $lessonSlug: String!
   $unitSlug: String!
+  $lessonCohort: String
```
Comment (k-huggs) [inline src/node-lib/curriculum-api-2023/queries/lessonOverview/lessonOveview.gql:5]: Should this be a required entry when using the api?

File: src/config/cohort.ts [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const NEW_COHORT = "2023-2024";
+export const LEGACY_COHORT = "2020-2023";
```
Comment (k-huggs) [inline src/config/cohort.ts:2]: I'm guessing that legacy cohorts may grow in the future, will this better as an array ? Not sure myself...

File: src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug].test.tsx [hunk 3]
```diff
@@ -90,18 +91,61 @@ describe("pages/teachers/lessons", () => {
     expect(screen.queryByTestId("download-all-button")).not.toBeInTheDocument();
   });
 
-  it("share button is disabled with non legacy content", async () => {
-    render(
+  it("share button is not disabled with non legacy content", () => {
```
Comment (k-huggs) [inline src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug].test.tsx:94]: The two tests have the same description, one passes null and the other `LEGACY_CONTENT` as lessonCohort, could potentially state the difference in them

**Review by k-huggs — COMMENTED** [review]
Couple of queries

File: src/config/cohort.ts (hunk 2 above)
Comment (JBR90) [reply src/config/cohort.ts:2]: Lets keep it as a variable until we need more

**Comment by JBR90** [comment]
@jack-skerrett-oak   getting 500 on the lesson overview pages, investigating - hold of review atm

File: src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug].test.tsx [hunk 4]
```diff
@@ -90,18 +91,61 @@ describe("pages/teachers/lessons", () => {
     expect(screen.queryByTestId("download-all-button")).not.toBeInTheDocument();
   });
 
-  it("share button is disabled with non legacy content", async () => {
-    render(
+  it("share button is not disabled with non legacy content (lessonCohort is null)", () => {
```
Comment (njblore) [inline src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug].test.tsx:94]: should this be 'legacy content' not 'non legacy content'? same below

**Review by njblore — APPROVED** [review]
Looks good, just a q with the test case titles

**Review by jack-skerrett-oak — APPROVED** [review]
Nothing seems to have changed

**Pull request merged**